Source/WebKit/UIProcess/AuxiliaryProcessProxy.cpp

```diff
@@ -89,6 +89,9 @@ AuxiliaryProcessProxy::AuxiliaryProcessProxy(ShouldTakeUIBackgroundAssertion sho
 
 AuxiliaryProcessProxy::~AuxiliaryProcessProxy()
 {
+    if (state() != State::Terminated)
+        platformStartConnectionTerminationWatchdog();
+
```

Comment:
I wonder if it's a viable option to call this in the NetworkProcessProxy destructor (and possibly the other proxy destructors), and avoid the ProcessAssertion changes, since the object will not be partially destroyed, then?

Reply:
The reason I am wondering this is that I think it may make sense to keep having the `ProcessAssertion` constructor take an `AuxiliaryProcessProxy&` parameter, since we normally only want to take assertions on these process types.

Reply:
I did leave in the `ProcessAssertion::create` and `ProcessAndUIAssertion::create` variants that take in an `AuxiliaryProcessProxy` as an argument since that's probably the main way you want to interact with those classes.

But calling `ProcessAndUIAssertion::create` from the NetworkProcessProxy destructor doesn't solve anything, since the problem with passing `this` associated with an object that's being destructed to another function is that the other function could inadvertently `ref()` it, which would cause a UAF. That doesn't actually happen the way the code is currently written, but I guess I am being pedantic and trying to avoid someone accidentally causing that to happen in the future.

Reply:
Ah, yes, that is a very good point, thanks!